### PR TITLE
Global msq limit implemented

### PR DESCRIPTION
**Pull request by nozjkoitop**

### CODE REVIEW

**Comment by vivek807** [comment]
Hi, 
This change will also require updates to the documentation, right?

**Comment by nozjkoitop** [comment]
> Hi, This change will also require updates to the documentation, right?

You're right, thanks for noticing, I've updated the docs

File: indexing-service/src/test/java/org/apache/druid/indexing/overlord/MSQControllerTaskLimitTest.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    Assert.assertEquals("no tasks should be known", 0, taskQueue.getTasks().size());
+    Assert.assertEquals("no tasks should be running", 0, taskQueue.getRunningTaskCount().size());
+
+    // Add all tasks.
+    for (int i = 0; i < numTasks; i++) {
+      final NoopTask testTask = createMsqTestTask(1000L);
+      taskQueue.add(testTask);
+    }
+
+    Thread.sleep(3000);
```
Comment (vivek807) [inline indexing-service/src/test/java/org/apache/druid/indexing/overlord/MSQControllerTaskLimitTest.java:96]: Is it possible to mark tasks complete manually. `Thread.sleep` would not be great idea here, I think

**Comment by vivek807** [comment]
`LGTM`

File: indexing-service/src/test/java/org/apache/druid/indexing/overlord/MSQControllerTaskLimitTest.java (hunk 1 above)
Comment (nozjkoitop) [reply indexing-service/src/test/java/org/apache/druid/indexing/overlord/MSQControllerTaskLimitTest.java:96]: This is more less how it's done in TaskQueueScaleTest, I'm not sure how to make it manageable as we are testing the way how taskQueue is calling runner.run()

Comment (vivek807) [reply indexing-service/src/test/java/org/apache/druid/indexing/overlord/MSQControllerTaskLimitTest.java:96]: Yes, I checked other places as well, and `Thread.sleep` is used there too. So I believe we’re good. For example:

```
private void waitUntil(KinesisIndexTask task, Predicate<KinesisIndexTask> predicate)
      throws InterruptedException
  {
    while (!predicate.test(task)) {
      Thread.sleep(10);
    }
  }
  ```

**Review by vivek807 — APPROVED** [review]
LGTM

**Pull request merged**